twine-components/src/example/area_calculators.rs

```diff
@@ -0,0 +1,100 @@
+use std::f64::consts::PI;
+
+use serde::{Deserialize, Serialize};
+use twine_core::Component;
+use uom::si::f64::{Area, Length};
+
+/// Component for calculating the area of a circle.
+pub struct CircleArea;
+
+/// Component for calculating the area of a rectangle.
+pub struct RectangleArea;
+
+/// Input structure for the `CircleArea` component.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CircleInput {
+    /// The radius of the circle.
+    pub radius: Length,
+}
+
+/// Input structure for the `RectangleArea` component.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct RectangleInput {
+    /// The length of the rectangle.
+    pub length: Length,
+    /// The width of the rectangle.
+    pub width: Length,
+}
+
+/// Output structure for `CircleArea` and `RectangleArea`.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Output {
+    /// The calculated area of the shape.
+    pub area: Area,
+}
+
+impl Component for CircleArea {
+    type Config = ();
+    type Input = CircleInput;
+    type Output = Output;
+
+    fn create(_config: Self::Config) -> impl Fn(Self::Input) -> Self::Output {
+        |input| Self::Output {
+            area: PI * input.radius * input.radius,
+        }
+    }
+}
+
+impl Component for RectangleArea {
+    type Config = ();
+    type Input = RectangleInput;
+    type Output = Output;
+
+    fn create(_config: Self::Config) -> impl Fn(Self::Input) -> Self::Output {
+        |input| Self::Output {
+            area: input.length * input.width,
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use approx::assert_relative_eq;
+    use uom::si::{
+        area::{square_centimeter, square_foot, square_mile},
+        length::{foot, inch, kilometer},
+    };
+
+    #[test]
+    fn circle_area_calculator() {
+        let circle_area_fn = CircleArea::create(());
```

Comment:
This is making me wonder if we should just get rid of `Config` and use the component structs themselves. Just use `new` and maybe add that `run` function that references `&self`. I realize this is me backtracking, but that seems like more idiomatic rust now that I'm looking at this.